### ISSUE TITLE
Internal link (from markdown) are normal link and trigger full page load

### ISSUE THREAD

**Issue by MoOx** [MoOx/statinamic#67]
That's a shame isn't it?
Before handling this, we need to work on #11 (cause this will handle markdown as react components instead of html).

**Comment by thangngoc89** [comment]
I think this can be solve seperately with #11 

I'm working on a branch which use [markdown-it-react-renderer](https://github.com/thangngoc89/markdown-it-react-renderer) to parse markdown from rawBody and replace `<a>` tag with `<Link>` component from react-router

https://github.com/MoOx/statinamic/compare/babel-node-loaders...thangngoc89:use-renderer?expand=1

**Comment by thangngoc89** [comment]
Maintaining a markdown it renderer is a big deal and a time-consuming task.

I chosed a different approach by using [html-to-react](https://www.npmjs.com/package/html-to-react) to parse html data from markdown-it and replace `<a>` and `<Link>` component. This is easier and faster.

1 down side is they are using lodash without cherry picking, so this will increase bundle size significantly. I tried to send them a PR later for this.

https://github.com/MoOx/statinamic/compare/babel-node-loaders...thangngoc89:html-to-react?expand=1

**Comment by MoOx** [comment]
basename as been introduced after I asked for it for statinamic, so maybe we should use it now :)

We will only need to to replace internal `<a` tags that matches the baseUrl (+pathname).

This code should definitely be in the core.

**Comment by thangngoc89** [comment]
Haha. Didn't know about that

**Comment by thangngoc89** [comment]
Is this a right approach?  Should I make an PR for this after big pr landed?

**Comment by MoOx** [comment]
Looks like a good start since diff seems to be small.
I just hope the html2react thing is not too big because as you made this, it will be in the client side.

**Comment by thangngoc89** [comment]
I raised an issue here https://github.com/mikenikles/html-to-react/issues/22 

html2react only have 2 deps: htmlparser2 and lodash (they only use 4-5 functions of lodash)

**Comment by MoOx** [comment]
Anyway I think on the long term we will need to handle this before client side (because we can).
But that's a start :)

**Comment by thangngoc89** [comment]
Ok. I'll send a PR then.

**Comment by thangngoc89** [comment]
Thanks to https://github.com/gatsbyjs/gatsby/issues/93

We can use this LoC to solve this issue:

```js
import catchLinks from 'catch-links'
import { browserHistory } from 'react-router'

if (typeof window !== 'undefined') {
  catchLinks(window, (href) => {
    browserHistory.push(href)
  })
}
```

But I would love to have a proper implementation of this using remark, remark-react :)

**Comment by thangngoc89** [comment]
'catch-links' have a bug with in page anchor. 

I have a fork of my own here : https://github.com/thangngoc89/blog/commits/master/web_modules/app/catchLinks.js

**Comment by thangngoc89** [comment]
Forgot to mention that, the above function only work with website that doesn't use pathname. 

For website that uses pathname (like docs page) you have to prepend pathname or use react-router basename

**Comment by MoOx** [comment]
Did you opened an issue on the original repo for this bug?

**Comment by thangngoc89** [comment]
I plant to send a PR, just forgot.

**Comment by MoOx** [comment]
react-router does that already on links it handle, so maybe we should do this in markdown content only?

**Comment by thangngoc89** [comment]
Yes. I'm doing that. https://github.com/thangngoc89/blog/blob/master/web_modules/app/routes.js#L10-L18

**Comment by MoOx** [comment]
We should add this logic in PageContainer to scope this in the dangerousHTML part.

**Comment by thangngoc89** [comment]
@MoOx Will add it in #165

**Comment by MoOx** [comment]
Closed by 07aaa7be263453b76e664792919b132ba7159db8
